### PR TITLE
Destination: AWS Event Bridge

### PR DESCRIPTION
**Pull request by dhawal1248**
## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ - ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ - ] My code follows the style guidelines of this project
- [ - ] I have performed a self-review of my own code
- [ - ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-server/410)

### CODE REVIEW

File: processor/processor_test.go [hunk 1]
```diff
@@ -1417,7 +1417,7 @@ func assertDestinationTransform(messages map[string]mockEventData, destinationID
 	return func(clientEvents []transformer.TransformerEventT, url string, batchSize int, breakIntoBatchWhenUserChanges bool) transformer.ResponseT {
 		destinationDefinitionName := expectations.destinationDefinitionName
 
-		Expect(url).To(Equal(fmt.Sprintf("http://localhost:9090/v0/%s?whSchemaVersion=", destinationDefinitionName)))
+		Expect(url).To(Equal(fmt.Sprintf("http://localhost:9090/v0/%s", destinationDefinitionName)))
```
Comment (arnab-p) [inline processor/processor_test.go:1420]: Do we need to change this?

File: services/streammanager/eventbridge/eventBridgeManager.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
+package eventbridge
```
Comment (arnab-p) [inline services/streammanager/eventbridge/eventBridgeManager.go:1]: Let's keep the file name as per convention of the repo? `eventbridgemanager.go`?

File: processor/processor_test.go (hunk 1 above)
Comment (dhawal1248) [reply processor/processor_test.go:1420]: done

File: services/streammanager/eventbridge/eventBridgeManager.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+func init() {
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destinationConfig interface{}) (eventbridge.EventBridge, error) {
+	config := Config{}
+
+	jsonConfig, err := json.Marshal(destinationConfig)
+	err = json.Unmarshal(jsonConfig, &config)
```
[the remark this answers is not in the document]
Comment (dhawal1248) [reply services/streammanager/eventbridge/eventBridgeManager.go:30]: done

File: services/streammanager/eventbridge/eventBridgeManager.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+type Config struct {
+	Region       string
+	AccessKeyID  string
+	AccessKey    string
+	EventBusName string
+	ResourceID   []map[string]interface{}
+	DetailType   string
+}
+
+func init() {
```
[the remark this answers is not in the document]
Comment (dhawal1248) [reply services/streammanager/eventbridge/eventBridgeManager.go]: done

File: services/streammanager/eventbridge/eventBridgeManager.go (hunk 2 above)
Comment (dhawal1248) [reply services/streammanager/eventbridge/eventBridgeManager.go:1]: done

**Pull request closed without merging**